src/3_mockDep.test.js

```diff
@@ -1,20 +1,11 @@
 import { generatePassword } from "./3_mockDep";
 
 describe("3_mockDep_test", () => {
-  
-  //  const password = jest.fn(() => validatePasswordApi("@83XD310256"));
-  // it("should resolve if password is truely", async () => {
-  //   await expect(password).toBeTruthy();
-  // });
-  // it("shuld be smaller than 8 wrods", async () => {
-  //   await expect(password.length()).toBeLessThanOrEqual(8);
-  // });
-  // it("shuld include special character", async () => {
-  //   const specialChars = /[`!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/;
-  //   await expect(result).toMatch(specialChars);
-  // });
-  // it("should have random number < 10", async () => {
-  //   const result = await jest.fn(() => randomizer(1, 10));
-  //   expect(result).toBeLessThan(10);
-  // });
+  it("should give you an error when the size is 0", async () => {
+    await expect(() => generatePassword(0, "")).toThrowError();
```

Comment:
todo/bug: Have you seen this test passing? If you run it, the output is [this error](https://github.com/TingChiehLin/testing_practice/assets/14869087/668f6503-686a-4eec-af36-6489a0f2fa92).

When testing async functions, the way we catch errors is different. [check the docs](https://github.com/TingChiehLin/testing_practice/assets/14869087/668f6503-686a-4eec-af36-6489a0f2fa92)